WordPress/gutenberg
# Rename blocks in the Editor via Inline Editing in List View

## What?


Adds the ability to give blocks (currently limited to Group block) a custom name in List View (_only_) via 

- a `Rename` item in the block's options menu
- double click on block name within list view.

This is then displayed in the List View (although that functionality was enabled in a previous PR).

⚠️ Unlike https://github.com/WordPress/gutenberg/pull/42605 this approach uses the `Popover` component to display the input _on top of_ the block name in the List View. This should help to offset the accessibility concerns around focus management.

Alternative to https://github.com/WordPress/gutenberg/pull/53735

Superceeds https://github.com/WordPress/gutenberg/pull/42605.

## Why?


Same reasons as https://github.com/WordPress/gutenberg/pull/53735. The only difference is that this PR explores the route of allowing "inline" renaming as @WordPress/gutenberg-design were keen on seeing this approach over the Modal based approach explored in https://github.com/WordPress/gutenberg/pull/53735.

## How?


Same method as in https://github.com/WordPress/gutenberg/pull/53735 but now:

- a `Popover` is used to display the renaming input.
- focus is transferred and managed


## Known Issues/Concerns

- currently the block options menu seems to auto-close when the `Popover` displays. This is not intentional but it means the focus doesn't end up back on the `Rename` menu option which might be confusing for non-visual users.
- there is no clear save/cancel UI when editing. This may confuse users who don't know about the `ENTER` to submit pattern. Suggestion is to perhaps use the `submit` icon on a button which shows on the right hand side of the input.

## Testing Instructions





- New Post (or enter the Site Editor)
- Add some blocks and then Group them.
- Open List View
- Now open the block options menu (either via List View or via the block in the canvas) and click `Rename`.
- Type a custom name and save.
- See custom name is now shown in List View.
- Bring up the `Rename` UI again, but this time remove all of the text and submit.
- See the value reset to the block's original name (e.g. `Group`).
- Save the change and verify it updated. 
- Switch to codeview and check the `metadata.name` attribute was reset (it should _not_ be `Group`).
- Now back in List View double click on the name of the block and see that you can rename like that.
- Check that single clicking on blocks in List View does not have a delay.

### Testing Instructions for Keyboard


- Prefer using a screenreader here as well.
- Same as above but ensure focus ends up where you expected.
- Ensure that you have sufficient context to understand where you are within the editor and the block name editing cycle.
- Check that controls make sense.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/444434/cc81aba2-16bb-4c58-93ec-173e069e6c10
